src/argilla/server/settings.py

```diff
@@ -127,7 +129,38 @@ def normalize_base_url(cls, base_url: str):
 
     @validator("database_url", always=True)
     def set_database_url_default(cls, database_url: str, values: dict) -> str:
-        return database_url or f"sqlite:///{os.path.join(values['home_path'], 'argilla.db')}?check_same_thread=False"
+        home_path = values.get("home_path")
+        sqlite_file = os.path.join(home_path, "argilla.db")
+        return database_url or f"sqlite+aiosqlite:///{sqlite_file}?check_same_thread=False"
+
+    @validator("database_url", pre=True)
+    def set_database_url(cls, database_url: str, values: dict) -> str:
+        if not database_url:
+            home_path = values.get("home_path")
+            sqlite_file = os.path.join(home_path, "argilla.db")
+            return f"sqlite+aiosqlite:///{sqlite_file}?check_same_thread=False"
```

Comment:
Do we need this? if the `set_database_url_default` validator is executed for all cases, for an unset database url the default validator should be enough

Reply:
my bad, I forgot to remove the old validator.